tests/test_routes.py

```diff
@@ -6,3 +9,52 @@ def test_index_page(flask_test_client):
 def test_upload_page(flask_test_client):
     response = flask_test_client.get("/upload")
     assert response.status_code == 200
+
+
+def test_upload_xlsx_successful(
+    requests_mock, example_pre_ingest_data_file, flask_test_client
+):
+    requests_mock.post(
+        "http://localhost:8080/ingest",
+        content=b'"Success: Spreadsheet data ingested"\n',
+    )
+    response = flask_test_client.post(
+        "/upload", data={"ingest_spreadsheet": example_pre_ingest_data_file}
+    )
+    page_html = BeautifulSoup(response.data)
+    assert response.status_code == 200
+    assert "Success: Spreadsheet data ingested" in str(page_html)
+
+
+def test_upload_xlsx_errors(
+    requests_mock, example_pre_ingest_data_file, flask_test_client
+):
+    requests_mock.post(
+        "http://localhost:8080/ingest",
+        content=b'{\n "validation_errors": [\n    "Wrong '
+        b'Type Failure: Sheet \\"Funding\\" column \\"Spend for Reporting '
+        b'Period\\" expected type \\"float64\\", got type \\"object\\"",\n'
+        b'"Enum Value Failure: Sheet \\"RiskRegister\\" Column '
+        b'\\"Pre-mitigatedImpact\\" Row 5 Value \\"Error\\" is not a valid '
+        b'enum value."]\n}\n',
+    )
+    response = flask_test_client.post(
+        "/upload", data={"ingest_spreadsheet": example_pre_ingest_data_file}
+    )
+    page_html = BeautifulSoup(response.data)
+    assert response.status_code == 200
+    assert (
+        'Wrong Type Failure: Sheet "Funding" column "Spend for Reporting '
+        'Period" expected type "float64", got type "object'
+    ) in str(page_html)
+    assert (
+        'Enum Value Failure: Sheet "RiskRegister" Column '
+        '"Pre-mitigatedImpact" Row 5 Value "Error" is not a valid enum value.'
+    ) in str(page_html)
+
+
+def test_upload_wrong_format(flask_test_client, example_ingest_wrong_format):
+    response = flask_test_client.post(
+        "/upload", data={"ingest_spreadsheet": example_ingest_wrong_format}
+    )
+    assert response.status_code == 200
```

Comment:
Can we assert on the page contents in this case also? Ideally it would include a friendly message to the user in this scenario?

Reply:
Yes, of course at the moment it will display a simple unknown file format message. I'm assuming all these messages will change once the design team get back to us with more detailed validation errors